backend/src/main/java/ca/gov/dtsstn/cdcp/api/service/UserService.java

```diff
@@ -90,16 +98,20 @@ public ConfirmationCode createConfirmationCodeForUser(String userId) {
 	}
 
 	public Subscription createSubscriptionForUser(String userId, Subscription subscription) {
+		log.debug("subscription {}", subscription.getLanguage());
 		Assert.hasText(userId, "userId is required; it must not be null or blank");
-		Assert.hasText(subscription.getAlertType().getCode(), "subscription.alertType.code is required; it must not be null or blank");
-		Assert.notNull(subscription.getPreferredLanguage(), "subscription.preferredLanguage is required; it must not be null");
+		Assert.notNull(Optional.ofNullable(subscription.getAlertType()).map(AlertType::getCode).orElse(null), "subscription.alertType.code is required; it must not be null or blank");
+		Assert.notNull(Optional.ofNullable(subscription.getLanguage()).map(Language::getMsLocaleCode).orElse(null), "subscription.getLanguage.getMsLocaleCode is required; it must not be null");
 
 		log.debug("Fetching user [{}] from repository", userId);
 		final var user = userRepository.findById(userId).orElseThrow();
 
 		log.debug("Fetching alert type [{}] from repository", subscription.getAlertType().getCode());
 		final var alertType = alertTypeRepository.findByCode(subscription.getAlertType().getCode()).orElseThrow();
 
+		log.debug("Fetching language [{}] from repository", subscription.getLanguage().getMsLocaleCode());
```

Comment:
```suggestion
		log.debug("Fetching language with msLocaleCode=[{}] from repository", subscription.getLanguage().getMsLocaleCode());
```

Reply:
(since there are more than one kinds of codes for this)

Reply:
updated